spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -98,13 +101,25 @@ public Subscriber subscribe(String subscription, MessageReceiver messageReceiver
 	@Override
 	public List<AcknowledgeablePubsubMessage> pull(String subscription, Integer maxMessages,
 			Boolean returnImmediately) {
+		Assert.hasText(subscription, "The subscription can't be null or empty.");
+
+		if (maxMessages != null) {
+			Assert.isTrue(maxMessages > 0, "The maxMessages must not be greater than 0.");
```

Comment:
So, should we provide the value "greater than 0" or not?

Reply:
Good catch.